src/client.js

```diff
@@ -0,0 +1,175 @@
+     * @param {string} clientDataCenter: needed to use DefaultLoadBalancingPolicy for load balancing the 
+     *                              requests to cassandra nodes.
+     * @param {Object} clientOptions: Client and query configuration options while setting up the connection. 
+     * @param {Object} emitter: to emit logs. 
+     */
+    constructor(clientName, keyspace, clientDataCenter, clientOptions, emitter) {
+        this.clientName = clientName;
+
+        this.defaultClientOptions = {
+            contactPoints: ['localhost'],
+            // required retry logic to be implemented in business logic
+            policies: {
+                // TODO: This policy neither retries or nor ignores. All the methods with this policy 
+                // throw exceptions. Thus, a retry policy needs to be added to the wrapper itself. 
+                retry: new cassandra.policies.retry.FallthroughRetryPolicy(),
+            },
+            queryOptions: {
+                prepare: true,
+            }
+        }
+        // common config defaults should go here.
+        this.clientOptions = Object.assign(
+            this.defaultClientOptions, 
+            clientOptions,
+            { 
+                applicationName: clientName,
+                keyspace,
+                localDataCenter: clientDataCenter,
+            }
+        );
+        this.emitter = emitter;
+        this.client = new cassandra.Client(this.clientOptions);
+    }
+
+    async connect() {
+        try {
+            this.client.on('log', (level, loggerName, message, furtherInfo) => {
+                if (level === 'verbose') {
+                    return;
+                }
+                this._logMessage(level, `${loggerName}: ${message}`, furtherInfo);
+             });
+             return await this.client.connect();
+        } catch (err) {
+            this._error(`Encountered error while connecting to cassandra node.`, err);
+            Promise.reject(err);
+        }
+    }
+
+    /**
+     * Executes a query on client.
+     * Example:
+     * const result = await client.execute(
+     *      'SELECT eid, name, joining_date, department FROM employees WHERE eid = ?', 
+     *      [10]
+     * );
+     * 
+     * @param {string} query -  query to be executed. 
+     * @param {Array | Object} params - params to be passed for parameterized query. 
+     * @param {Object} queryOptions - per query execution options  
+     * @returns 
+     */
+
+    async execute(query, params, queryOptions) {
+        try {
+            const result = await this.client.execute(query, params, queryOptions);
+            this._success('Successfully executed the query.');
+            return result;
+        } catch (err) {
+            this._error(`Encountered error while executing query "${err.query}" on coordinator "${err.coordinator}".\n`, err.stack);
+            Promise.reject(err);
```

Comment:
return the promise or throw the error
```suggestion
            return Promise.reject(err);
```